src/deploy/functions/services/firestore.ts

```diff
@@ -0,0 +1,25 @@
+import * as backend from "../backend";
+import * as firestore from "../../../gcp/firestore";
+import { FirebaseError } from "../../../error";
+
+/**
+ * Sets a firestore event trigger's region to the firestore database region.
+ * @param endpoint the firestore endpoint
+ */
+export async function ensureFirestoreTriggerRegion(
+  endpoint: backend.Endpoint & backend.EventTriggered
+): Promise<void> {
+  const db = await firestore.getDatabase(
+    endpoint.project,
+    endpoint.eventTrigger.eventFilters?.database || "(default)"
+  );
+  const dbRegion = db.locationId;
+  if (!endpoint.eventTrigger.region) {
+    endpoint.eventTrigger.region = dbRegion;
```

Comment:
Q: Aren't Firestore db location multi-region? (e.g. `nam5` vs `us-central1`)? Is it possible to have multi-region triggers?

Reply:
Ah ha sounds like it should be possible, just like storage regions:

https://cloud.google.com/eventarc/docs/locations#gcs-triggers

Reply:
Should we be following GCS implementation and do more granular checks (e.g. is this region in this multiregion):

https://github.com/firebase/firebase-tools/blob/3a6e0f05e544da914dcf900d6d149aeed5068d9f/src/deploy/functions/services/storage.ts#L58-L61

or is Firestore implementation much more straighforward?

Reply:
Yep Firestore can be multi-region triggers. To my knowledge, Firestore is more straightforward and just requires the trigger to be in the exact same region as the db. I haven't seen any documentation that function regions are limited like storage and should be fine to live in a location outside of the multi-region that Eventarc/Firestore lives

Reply:
As written, we'd reject trigger in `us-central1` because that's not the same as `nam5` even though it is techincally part of `nam5`. Is that what we want?

Reply:
resolved offline